create-convex-hull-around-points/src/main/java/com/esri/arcgismaps/sample/createconvexhullaroundpoints/MainActivity.kt

```diff
@@ -0,0 +1,219 @@
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.esri.arcgismaps.sample.createconvexhullaroundpoints.databinding.ActivityMainBinding
+import com.google.android.material.snackbar.Snackbar
+import kotlinx.coroutines.launch
+
+class MainActivity : AppCompatActivity() {
+
+    private val TAG = MainActivity::class.java.simpleName
+
+    // set up data binding for the activity
+    private val activityMainBinding: ActivityMainBinding by lazy {
+        DataBindingUtil.setContentView(this, R.layout.activity_main)
+    }
+
+    // setup binding for the MapView
+    private val mapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    // action button that creates the canvas hull
+    private val createButton by lazy {
+        activityMainBinding.createButton
+    }
+
+    // action button to reset the map
+    private val resetButton by lazy {
+        activityMainBinding.resetButton
+    }
+
+    // a red marker symbol for points
+    private val pointSymbol = SimpleMarkerSymbol(SimpleMarkerSymbolStyle.Circle, Color.red, 10f)
+
+    // a blue line symbol
+    private val lineSymbol = SimpleLineSymbol(SimpleLineSymbolStyle.Solid, Color.blue, 3f)
+
+    // a fill symbol with an empty fill for polygons
+    private val fillSymbol = SimpleFillSymbol(SimpleFillSymbolStyle.Null, Color.red, lineSymbol)
+
+    // set up the point graphic with point symbol
+    private val pointGraphic = Graphic(symbol = pointSymbol)
+
+    // init the convex hull graphic
+    private val convexHullGraphic = Graphic()
+
+    // create a graphics overlay to draw all graphics
+    private val graphicsOverlay = GraphicsOverlay()
+
+    // list to store the selected map points
+    private val inputPoints = mutableListOf<Point>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // add point and convex hull graphics to the graphics overlay
+        graphicsOverlay.graphics.addAll(listOf(pointGraphic, convexHullGraphic))
+
+        // create and add a map with topographic basemap style
+        val map = ArcGISMap(BasemapStyle.ArcGISTopographic).apply {
+            // set a default initial point and scale
+            initialViewpoint = Viewpoint(Point(34.77, -10.24), 20e7)
+        }
+
+        // configure map view assignments
+        mapView.apply {
+            this.map = map
+            // add the graphics overlay to the mapview
+            graphicsOverlays.add(graphicsOverlay)
+        }
+
+        lifecycleScope.launch {
+            // if the map load fails show the error and return
+            map.load().onFailure {
+                showError("Error loading map")
+                return@launch
```

Comment:
```suggestion
                return@launch showError("Error loading map")
```
Could be condensed here.